Task BJS2-73134

FROLOV-KI opened the pull request:
1) Player class added

FROLOV-KI committed:
Task BJS2-73134

1) Player class added

MarselYsup commented on src/main/java/school/faang/spotify/Player.java:
```diff
@@ -0,0 +1,70 @@
+    public synchronized void skip() {
+        synchronized (lock) {
+            log.info("skipping {} track", tracks.get(songIndex));
+            songIndex++;
```
надо проверять, что ты не вышел за лист, чтобы не получить exception выход за массив - надо менять index но в рамках размера листа

MarselYsup commented on src/main/java/school/faang/spotify/Player.java:
```diff
@@ -0,0 +1,70 @@
+
+    public synchronized void pause() {
+        synchronized (lock) {
+            isPlaying = false;
```
думаю стоит проверять играет ли музыка, чтобы ее запаузить и наоборот на стороне запуска трека

MarselYsup commented on src/main/java/school/faang/spotify/Main.java:
```diff
@@ -0,0 +1,37 @@
+                player::play
+        );
+
+        ExecutorService executor = Executors.newFixedThreadPool(3);
```
Количество потоков лучше вынести в константу для улучшения читаемости и возможности изменения.

MarselYsup commented on src/main/java/school/faang/spotify/Main.java:
```diff
@@ -0,0 +1,37 @@
+        ExecutorService executor = Executors.newFixedThreadPool(3);
+        users.forEach(executor::submit);
+        executor.shutdown();
+        if (!executor.awaitTermination(5, TimeUnit.SECONDS)) {
```
время ожидания тоже лучше вынести в константу, лучше не оставлять магических чисел

FROLOV-KI committed:
PR issues resolved

1) Check track list boundaries added
2) Player play\pause logic updated
3) Magic numbers resolved

MarselYsup commented:
Огонь, круто получилось 🔥
